addons/website_rating_project/models/project.py

```diff
@@ -23,3 +23,108 @@ def action_view_all_rating(self):
+            '1 month': {'happy': 0, 'avg': 0, 'unhappy': 0, 'total': 0},
+            '3 month': {'happy': 0, 'avg': 0, 'unhappy': 0, 'total': 0},
+        } for rated_partner_id in set(partner_ids)]
+
+        def increment_rating(partner, days, rating):
+            if rating == 10:
+                partner[days]['happy'] += 1
+            if rating == 5:
+                partner[days]['avg'] += 1
+            if rating == 1:
+                partner[days]['unhappy'] += 1
+            partner[days]['total'] += 1
+
+        for record in all_record:
+            partner = [x for x in partner_ratings if x['rated_partner_id'].id == record['rated_partner_id']][0]
+            if record['days'] == 'week':
+                for rating in record['rating']:
+                    increment_rating(partner, '15 days', rating)
+                    increment_rating(partner, '1 month', rating)
+                    increment_rating(partner, '3 month', rating)
+            if record['days'] == '15 days':
+                for rating in record['rating']:
+                    increment_rating(partner, '1 month', rating)
+                    increment_rating(partner, '3 month', rating)
+            if record['days'] == '1 month':
+                for rating in record['rating']:
+                    increment_rating(partner, '3 month', rating)
+
+        for record in all_record:
+            rating_ids += record['rating_ids']
+            partner = [x for x in partner_ratings if x['rated_partner_id'].id == record['rated_partner_id']][0]
+            for rating in record['rating']:
+                increment_rating(partner, record['days'], rating)
+
+        statistic = {
+            'week': {'avg': 0, 'happy': 0, 'unhappy': 0, 'total': 0},
+            '1 month': {'avg': 0, 'happy': 0, 'unhappy': 0, 'total': 0},
+            '3 month': {'avg': 0, 'happy': 0, 'unhappy': 0, 'total': 0},
+        }
+        days = ['week', '1 month', '3 month']
+        ratings = ['avg', 'happy', 'unhappy']
+        matrix = [(x, y) for x in days for y in ratings]
+        for partner in partner_ratings:
+            for day, rating in matrix:
+                statistic[day][rating] += partner[day][rating]
+                statistic[day]['total'] += partner[day][rating]
+        for day, rating in matrix:
+            total = statistic[day]['total']
+            percentage = 'percentage_%s' % rating
+            statistic[day][percentage] = total and "%.2f" % (( int(statistic[day][rating] * 100 / float(total) *100) /100 )) or 0.0
+
+        partner_ratings = sorted(partner_ratings, key=lambda k: k['15 days']['total'], reverse=True)[:5]
+        for rating in partner_ratings:
+            ratring = rating['15 days']
+            total = ratring['happy'] * 100 + ratring['avg'] * 50
+            rating['15 days']['%'] = total and "%.2f" % (total/ratring['total']) or 0.0
+
+        return {
+            'partner_rating': partner_ratings,
+            'statistic': statistic,
```

Comment:
`statistic` can mean everything and nothing. `periodical_stat` is more explicit IMHO